x-pack/legacy/plugins/advanced_ui_actions/public/customize_time_range_modal.tsx

```diff
@@ -0,0 +1,176 @@
+    const parentPanels = parent!.getInput().panels;
+
+    // Remove any explicit input to this child from the parent.
+    parent!.updateInput({
+      panels: {
+        ...parentPanels,
+        [embeddable.id]: {
+          ...parentPanels[embeddable.id],
+          explicitInput: {
+            ...parentPanels[embeddable.id].explicitInput,
+            timeRange: undefined,
+          },
+        },
+      },
+    });
+
+    this.props.onClose();
+  };
+
+  public render() {
+    return (
+      <React.Fragment>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle data-test-subj="customizePanelTitle">
+            {i18n.translate('xpack.advancedUiActions.customizeTimeRange.modal.headerTitle', {
+              defaultMessage: 'Customize panel time range',
+            })}
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          {' '}
+          <EuiFormRow
+            label={i18n.translate(
+              'xpack.advancedUiActions.customizePanelTimeRange.modal.optionsMenuForm.panelTitleFormRowLabel',
+              {
+                defaultMessage: 'Time range',
+              }
+            )}
+          >
+            <EuiSuperDatePicker
+              start={this.state.timeRange ? this.state.timeRange.from : undefined}
+              end={this.state.timeRange ? this.state.timeRange.to : undefined}
+              isPaused={false}
+              onTimeChange={this.onTimeChange}
+              showUpdateButton={false}
+              dateFormat={this.props.dateFormat}
+            />
+          </EuiFormRow>
+        </EuiModalBody>
+        <EuiModalFooter>
+          <EuiFlexGroup justifyContent="spaceBetween">
+            <EuiFlexItem>
+              <EuiFlexGroup gutterSize="s">
+                <EuiButtonEmpty
+                  onClick={this.inheritFromParent}
+                  color="danger"
+                  data-test-subj="removePerPanelTimeRangeButton"
+                  disabled={this.state.inheritTimeRange}
+                >
+                  {' '}
+                  {i18n.translate(
+                    'xpack.advancedUiActions.customizePanelTimeRange.modal.removeButtonTitle',
+                    {
+                      defaultMessage: 'Remove',
+                    }
+                  )}
+                </EuiButtonEmpty>
+              </EuiFlexGroup>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFlexGroup gutterSize="s">
+                <EuiButtonEmpty
+                  onClick={this.cancel}
+                  data-test-subj="cancelPerPanelTimeRangeButton"
+                >
+                  {' '}
+                  {i18n.translate(
+                    'xpack.advancedUiActions.customizePanelTimeRange.modal.cancelButtonTitle',
+                    {
+                      defaultMessage: 'Cancel',
+                    }
+                  )}
+                </EuiButtonEmpty>
+
+                <EuiButton
```

Comment:
Ok, I see the FlexGroup issue here. This is how it should be laid out:

```jsx
<EuiFlexGroup responsive={false}>
  <EuiFlexItem>
     ...remove button
  </EuiFlexItem>
  <EuiFlexItem grow={false}>
    ...cancel button
  </EuiFlexItem>
  <EuiFlexItem grow={false}>
    ...add button
  </EuiFlexItem>
</EuiFlexGroup>
```

Reply:
This flex group stuff still needs to be cleaned up. There's too many nested FlexGroups. You can use the code above for the layout.

Reply:
whoops, take 2, I think I did it right this time!